pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/GreatestTransformFunction.java

```diff
@@ -74,6 +135,35 @@ public float[] transformToFloatValuesSV(ValueBlock valueBlock) {
     return _floatValuesSV;
   }
 
+  @Override
+  public Pair<float[], RoaringBitmap> transformToFloatValuesSVWithNull(ValueBlock valueBlock) {
+    int numDocs = valueBlock.getNumDocs();
+    initFloatValuesSV(numDocs);
+    Pair<float[], RoaringBitmap> values = _arguments.get(0).transformToFloatValuesSVWithNull(valueBlock);
+    System.arraycopy(values.getLeft(), 0, _floatValuesSV, 0, numDocs);
+    RoaringBitmap nullBitmap = values.getRight();
+    for (int i = 1; i < _arguments.size(); i++) {
```

Comment:
Can we extract out this logic in a method. Seems like the only difference is the datatype of the array.